development/development/css_selectors.rst

```diff
@@ -0,0 +1,256 @@
+
+Location Independence
+~~~~~~~~~~~~~~~~~~~~~
+
+Given the ever-changing nature of most UI projects, and the move to more component-based architectures, it is in our interests not to style things based on where they are, but on what they are. That is to say, our components’ styling should not be reliant upon where we place them—they should remain entirely location independent.
+
+Let’s take an example of a call-to-action button that we have chosen to style via the following selector:
+
+.. code:: css
+
+    .promo a { }
+
+Not only does this have poor Selector Intent—it will greedily style any and every link inside of a ``.promo`` to look like a button—it is also pretty wasteful as a result of being so locationally dependent: we can’t reuse that button with its correct styling outside of ``.promo`` because it is explicitly tied to that location. A far better selector would have been:
+
+.. code:: css
+
+    .btn { }
+
+This single class can be reused anywhere outside of ``.promo`` and will always carry its correct styling. As a result of a better selector, this piece of UI is more portable, more recyclable, doesn’t have any dependencies, and has much better Selector Intent. **A component shouldn’t have to live in a certain place to look a certain way.**
+
+Portability
+~~~~~~~~~~~
+
+Reducing, or, ideally, removing, location dependence means that we can move components around our markup more freely, but how about improving our ability to move classes around components? On a much lower level, there are changes we can make to our selectors that make the selectors themselves—as opposed to the components they create—more portable. Take the following example:
+
+.. code:: css
+
+    input.btn { }
+
+This is a *qualified* selector; the leading ``input`` ties this ruleset to only being able to work on ``input`` elements. By omitting this qualification, we allow ourselves to reuse the ``.btn`` class on any element we choose, like an ``a``, for example, or a ``button``.
+
+Qualified selectors do not lend themselves well to being reused, and every selector we write should be authored with reuse in mind.
+
+Of course, there are times when you may want to legitimately qualify a selector—you might need to apply some very specific styling to a particular element when it carries a certain class, for example:
+
+.. code:: css
+
+    /**
+     * Embolden and color any element with a class of `.error`.
+     */
+    .error {
+        color: red;
+        font-weight: bold;
+    }
+
+    /**
+     * If the element is a `div`, also give it some box-like styling.
+     */
+    div.error {
+        padding: 10px;
+        border: 1px solid;
+    }
+
+This is one example where a qualified selector might be justifiable, but I would still recommend an approach more like:
```

Comment:
Try not to use I/we: `, but an approach like the following is still recommended:`

Reply:
Although I have to admit that there is probably a lot more that could be changed to get rid of I/we